### ISSUE TITLE
Detect redundant enumerate

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1825]
# Rule request

## Thesis

```python
# bad
for _, item in enumerate(items):
  ...

# good
for item in items:
  ...
```

## Reasoning

Simplification.

P.S. I'm not making it, have seen it in the wild 🙃

**Comment by orsinium** [comment]
The same can be applied to `dict.items` but it apparently requires to know the right type for the iterator. Probably, one more good candidate for typed-linter.

```python
for _, item in dctnr.items():
  ...
```

**Comment by sobolevn** [comment]
I will leave this one for `typed-linter`.

**Comment by orsinium** [comment]
It will be false-positive only if you have shadowed `enumerate` function, and I'd say it is a bad practice on its own and you deserve this false-positive then 🙃

**Comment by sobolevn** [comment]
Yes, my bad!

**Comment by ujju20** [comment]
Hello, I am a beginner in the Open source world and I found this issue to be good first so I would like to take this issue. So please guide me.

**Comment by sobolevn** [comment]
Hi @ujju20!

Make sure to read [our contributing guides](https://github.com/wemake-services/wemake-python-styleguide/blob/master/CONTRIBUTING.md)

Next, you will need to create a new `violation` and a new `visitor`, see our glossary what these things mean.

I am here to help: ask anything you need.

**Comment by lensvol** [comment]
If this is not taken by anybody, can I take it over?

**Comment by orsinium** [comment]
Sure, go on :)
